storm-core/src/jvm/org/apache/storm/utils/ConfigUtils.java

```diff
@@ -0,0 +1,711 @@
+    }
+
+    // public static mkStatsSampler // depends on Utils.evenSampler() TODO, this is sth we need to do after util
+    // public static readDefaultConfig // depends on Utils.clojurifyStructure and Utils.readDefaultConfig // TODO
+    // validate-configs-with-schemas is just a wrapper of ConfigValidation.validateFields(conf)
+
+    //For testing only
+    // for java
+    // try (SetMockedStormConfig mocked = new SetMockedStormConfig(conf)) {
+    //    run test ...
+    // }
+    //
+    // for clojure
+    // (with-open [mock (SetMockedStormConfig. conf)]
+    //     run test ...)
+    public static class SetMockedStormConfig implements Closeable {
+        public SetMockedStormConfig(Map conf) {
+            mockedStormConfig = conf;
+        }
+
+        @Override
+        public void close() {
+            mockedStormConfig = null;
+        }
+    }
+    private static Map mockedStormConfig = null;
+    public static Map readStormConfig() {
+        if (mockedStormConfig != null) return mockedStormConfig;
+        Map conf = Utils.readStormConfig();
+        ConfigValidation.validateFields(conf);
+        return conf;
+    }
+
+    public static Map readYamlConfig(String name, boolean mustExist) {
+        Map conf = Utils.findAndReadConfigFile(name, mustExist);
+        ConfigValidation.validateFields(conf);
+        return conf;
+    }
+
+    public static Map readYamlConfig(String name) {
+        return  readYamlConfig(name, true);
+    }
+
+    public static String absoluteStormLocalDir(Map conf) {
+        String stormHome = System.getProperty("storm.home");
+        String localDir = (String) conf.get(Config.STORM_LOCAL_DIR);
+        if (localDir == null) {
+            return (stormHome + FILE_SEPARATOR + "storm-local");
+        } else {
+            if (new File(localDir).isAbsolute()) {
+                return localDir;
+            } else {
+                return (stormHome + FILE_SEPARATOR + localDir);
+            }
+        }
+
+    }
+
+    public static String absoluteHealthCheckDir(Map conf) {
+        String stormHome = System.getProperty("storm.home");
+        String healthCheckDir = String.valueOf(conf.get(Config.STORM_HEALTH_CHECK_DIR));
+        if (healthCheckDir.equals("null")) {
+            return (stormHome + FILE_SEPARATOR + "healthchecks");
+        } else {
+            if (new File(healthCheckDir).isAbsolute()) {
+                return healthCheckDir;
+            } else {
+                return (stormHome + FILE_SEPARATOR + healthCheckDir);
+            }
+        }
+    }
+
+    public static String masterLocalDir(Map conf) throws IOException {
+        String ret = String.valueOf(conf.get(Config.STORM_LOCAL_DIR)) + FILE_SEPARATOR + "nimbus";
+        try {
+            FileUtils.forceMkdir(new File(ret));
+        } catch (IOException e) {
+            LOG.error("Failed to create dir " + ret, e);
+            throw e;
```

Comment:
Catching an exception to log it an rethrow it is not typically good practice.  Could we remove the catch/rethrow?  I also am confused why we do a mkdir here, but that is what the original code did so it is a good translation.  Could we file a JIRA to clean this up later?  Creating directories does not feel like something ConfigUtils should be doing.